java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspServerTelemetryManager.java

```diff
@@ -0,0 +1,167 @@
+        JsonArray prjProps = new JsonArray();
+
+        Map<String, Project> mp = prjs.stream()
+                .collect(Collectors.toMap(project -> project.getProjectDirectory().getPath(), project -> project));
+
+        for (FileObject workspaceFolder : workspaceClientFolders) {
+            try {
+                JsonObject obj = new JsonObject();
+                String prjPath = workspaceFolder.getPath();
+                String prjId = this.getPrjId(prjPath);
+                obj.addProperty("id", prjId);
+
+                // In future if different JDK is used for different project then this can be updated 
+                obj.addProperty("javaVersion", System.getProperty("java.version"));
+
+                if (mp.containsKey(prjPath)) {
+                    Project prj = mp.get(prjPath);
+
+                    String buildToolName = prj.getClass().getSimpleName();
+                    obj.addProperty("buildTool", (buildToolName.equals("NbGradleProjectImpl") ? "GradleProject" : "MavenProject"));
+
+                    obj.addProperty("openedWithProblems", ProjectProblems.isBroken(prj));
+
+                    boolean isPreviewFlagEnabled = this.isEnablePreivew(prj.getProjectDirectory(), buildToolName);
+                    obj.addProperty("enablePreview", isPreviewFlagEnabled);
+                } else {
+                    obj.addProperty("buildTool", this.STANDALONE_PRJ);
+                    obj.addProperty("javaVersion", System.getProperty("java.version"));
+                    obj.addProperty("openedWithProblems", false);
+
+                    boolean isPreviewFlagEnabled = this.isEnablePreivew(workspaceFolder, this.STANDALONE_PRJ);
+                    obj.addProperty("enablePreview", isPreviewFlagEnabled);
+                }
+
+                prjProps.add(obj);
+
+            } catch (NoSuchAlgorithmException ex) {
+                Exceptions.printStackTrace(ex);
+            }
+        }
+
+        properties.add("prjsInfo", prjProps);
+
+        properties.addProperty("timeToOpenPrjs", timeToOpenPrjs);
+        properties.addProperty("numOfPrjsOpened", workspaceClientFolders.size());
+        properties.addProperty("lspServerInitializationTime", System.currentTimeMillis() - this.lspServerIntiailizationTime);
+
+        this.sendTelemetry(new TelemetryEvent(MessageType.Info.toString(), this.WORKSPACE_INFO_EVT, properties));
+    }
+    
+    private boolean isEnablePreivew(FileObject source, String prjType) {
+        if (prjType.equals(this.STANDALONE_PRJ)) {
+            NbCodeLanguageClient client = Lookup.getDefault().lookup(NbCodeLanguageClient.class);
+            if (client == null) {
+                return false;
+            }
+            AtomicBoolean isEnablePreviewSet = new AtomicBoolean(false);
+            ConfigurationItem conf = new ConfigurationItem();
+            conf.setSection(client.getNbCodeCapabilities().getAltConfigurationPrefix() + "runConfig.vmOptions");
```

Comment:
I think that rather digging in client's configuration (which is an implementation-dependent thing), we should extend `NbCodeCapabilities` with the preview flag. Also eliminates the round-trip to the client.

Reply:
I like this idea but I am not sure how can we achieve this on the frontend side i.e. if it is a maven project then should we parse POM file to check whether preview flag is there or not? Can you please suggest something if you have some idea around this?